Marlin/src/module/stepper.cpp

```diff
@@ -2910,23 +2910,22 @@ hal_timer_t Stepper::block_phase_isr() {
       typedef struct {
         xy_long_t buffer[IS_COMPENSATION_BUFFER_SIZE];
         uint16_t index;
-        void add(const xy_long_t &input) {
+        FORCE_INLINE void add(const xy_long_t &input) {
           buffer[index] = input;
           if (++index == IS_COMPENSATION_BUFFER_SIZE) index = 0;
         }
+        FORCE_INLINE xy_long_t past_item(const uint16_t n) {
+          const uint16_t i = index + IS_COMPENSATION_BUFFER_SIZE - n;
+          return buffer[i < IS_COMPENSATION_BUFFER_SIZE ? i : i - IS_COMPENSATION_BUFFER_SIZE];
```

Comment:
I think we can save one operation here:

If (index >= n) return index - n; // 2 ops total
Else return  index + SIZE - n; // 3 ops total

Instead of 3 and 4

Reply:
Subtraction is required to determine "greater than" state anyway, so…
```cpp
const int16_t i = int16_t(index) - n;
return buffer[i > 0 ? i : i + IS_COMPENSATION_BUFFER_SIZE];
```